app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/UpdateCartItems.php

```diff
@@ -73,7 +73,7 @@ public function processCartItems(Quote $cart, array $items): void
             $quantity = (float)$item['quantity'];
 
             if ($quantity <= 0.0) {
-                $this->cartItemRepository->deleteById((int)$cart->getId(), $itemId);
+                $cart->removeItem($itemId);
```

Comment:
After further analysis, your point is valid. The current GraphQL implementation using `CartItemRepositoryInterface::deleteById()` is inconsistent with the rest of Magento (REST API, Controllers) which all use `Quote::removeItem()`.

But please add a comment explaining the fix.